pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -354,33 +357,39 @@ func (h *handler) OnRemove(key string, obj runtime.Object) (runtime.Object, erro
 		return obj, generic.ErrSkip
 	}
 
-	clusterName := infraObj.meta.GetLabels()[capi.ClusterLabelName]
+	clusterName := infra.meta.GetLabels()[capi.ClusterLabelName]
 	if clusterName == "" {
 		return obj, fmt.Errorf("error retrieving the clustername for machine, label key %s does not appear to exist for dynamic machine %s", capi.ClusterLabelName, key)
 	}
 
-	machine, err := rke2.GetMachineByOwner(h.machines, infraObj.meta)
-	if err != nil && !errors.Is(err, rke2.ErrNoMachineOwnerRef) {
+	machine, err := rke2.GetOwnerCAPIMachine(obj, h.machineCache)
+	if apierrors.IsNotFound(err) {
+		logrus.Debugf("[rkemachine] %s/%s: waiting: machine to be set as owner reference", infra.meta.GetNamespace(), infra.meta.GetName())
+		h.EnqueueAfter(infra, 10*time.Second)
+		return obj, generic.ErrSkip
+	}
+	if err != nil {
+		logrus.Errorf("[rkemachine] %s/%s: error getting machine by owner reference %v", infra.meta.GetNamespace(), infra.meta.GetName(), err)
 		return obj, err
 	}
 
-	if machine == nil || machine.Status.NodeRef == nil {
+	if machine.Status.NodeRef == nil {
 		// Machine noderef is nil, we should just allow deletion.
-		logrus.Debugf("[MachineProvision] There was no associated K8s node with this etcd dynamicmachine %s. Proceeding with deletion", key)
-		return h.doRemove(infraObj)
+		logrus.Debugf("[rkemachine] There was no associated K8s node with this machine %s. Proceeding with deletion", key)
```

Comment:
I know that this controller isn't registered as the `rke-machine` controller, and I would've updated it to be named that, but:

1. IMO it should've been named that originally
2. I only didn't update it to avoid finalizer migration.

I see 3 potential options:

1. Leave log message prefix as `rkemachine` and handler registration as `machine-provision`
2. Rename handlers and package to `rkemachine` to be in line with the other controllers in this package and do finalizer migration somewhere.
3. Convert log message prefix to `machineprovision`.

Reply:
I would be OK with #1 for now, given the caveat that we know how to grok for the logs for this controller.